src/services/itemLogin/index.ts

```diff
@@ -96,7 +97,21 @@ const plugin: FastifyPluginAsync = async (fastify) => {
       const member = notUndefined(user?.account);
       assertIsMember(member);
       return db.transaction(async (manager) => {
-        return itemLoginService.put(member, buildRepositories(manager), itemId, type);
+        const repositories = buildRepositories(manager);
+
+        const item = await itemService.get(member, repositories, itemId, PermissionLevel.Admin); // Validate permissions
+        const schema = await itemLoginService.getOneByItem(repositories, item.id);
+        if (schema) {
+          // If exists, then update the existing one
+          return itemLoginService.update(
+            repositories,
+            schema.id,
+            type ?? ItemLoginSchemaType.Username,
+          );
+        } else {
+          // If not exists, then create a new one
+          return itemLoginService.create(repositories, item.path, type);
```

Comment:
Should we await here?

```suggestion
          return await itemLoginService.create(repositories, item.path, type);
```